profiler/src/ProfilerEngine/Datadog.Profiler.Native.Windows/CrashReportingWindows.cpp

```diff
@@ -0,0 +1,299 @@
+            return 1;
+        }
+
+        SymInitialize(_process, nullptr, TRUE);
+
+        _modules = GetModules();
+    }
+
+    return result;
+}
+
+std::vector<std::pair<int32_t, std::string>> CrashReportingWindows::GetThreads()
+{
+    std::vector<std::pair<int32_t, std::string>> threads;
+
+    auto threadSnapshot = CreateToolhelp32Snapshot(TH32CS_SNAPTHREAD, _pid);
+
+    if (threadSnapshot == INVALID_HANDLE_VALUE)
+    {
+        return threads;
+    }
+
+    THREADENTRY32 threadEntry = {};
+    threadEntry.dwSize = sizeof(THREADENTRY32);
+
+    if (Thread32First(threadSnapshot, &threadEntry))
+    {
+        do
+        {
+            if (threadEntry.th32OwnerProcessID == _pid)
+            {
+                auto thread = OpenThread(THREAD_QUERY_INFORMATION, FALSE, threadEntry.th32ThreadID);
+
+                if (thread)
+                {
+                    std::string threadName;
+                    PWSTR description;
+                    
+                    if (SUCCEEDED(GetThreadDescription(thread, &description)))
+                    {
+                        threadName = shared::ToString(description);
+                    }
+
+                    threads.push_back({ threadEntry.th32ThreadID, threadName });
+
+                    CloseHandle(thread);
+                }                
+            }
+        } while (Thread32Next(threadSnapshot, &threadEntry));
+    }
+
+    CloseHandle(threadSnapshot);
+
+    return threads;
+}
+
+std::vector<StackFrame> CrashReportingWindows::GetThreadFrames(int32_t tid, ResolveManagedCallstack resolveManagedCallstack, void* callbackContext)
+{
+    std::vector<StackFrame> frames;
+
+    // Get the managed callstack
+    ResolveMethodData* managedCallstack;
+    int32_t numberOfManagedFrames;
+
+    auto resolved = resolveManagedCallstack(tid, callbackContext, &managedCallstack, &numberOfManagedFrames);
+
+    std::vector<StackFrame> managedFrames;
+
+    if (resolved == 0 && numberOfManagedFrames > 0)
+    {
+        managedFrames.reserve(numberOfManagedFrames);
+
+        for (int i = 0; i < numberOfManagedFrames; i++)
+        {
+            auto const& managedFrame = managedCallstack[i];
+
+            StackFrame stackFrame;
+            stackFrame.ip = managedFrame.ip;
+            stackFrame.sp = managedFrame.sp;
+            stackFrame.method = std::string(managedFrame.symbolName);
+            stackFrame.moduleAddress = managedFrame.moduleAddress;
+            stackFrame.symbolAddress = managedFrame.symbolAddress;
+            stackFrame.isSuspicious = managedFrame.isSuspicious;
+
+            managedFrames.push_back(std::move(stackFrame));
+        }
+    }
+
+    CONTEXT context = {};
+    context.ContextFlags = CONTEXT_FULL;
+
+    auto thread = OpenThread(THREAD_GET_CONTEXT, FALSE, tid);
+
+    if (thread == NULL)
+    {
+        return managedFrames;
+    }
+
+    if (GetThreadContext(thread, &context))
+    {
+        STACKFRAME_EX nativeStackFrame = {};
+#ifdef _M_X64
```

Comment:
_Technically_ we can also build for arm64... we have it all ready to go in CI and everything

Reply:
At this point I don't think we care about Windows on ARM64

Reply:
That's fine, it was only if it was a simple tweak👍